From 22689b02bb379d8e65385af4b43dd5e9591f21b7 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Sat, 29 Sep 2012 15:30:26 +0800
Subject: [PATCH 348/498] ENGR00225698 wifi: port atheros wifi sdk

modify compat-wireless to work at imx, maily the following changes:
1. chmod a+x for check_config.sh gen-compat-autoconf.sh
2. add freescale copyright for those modified files.
3. modify export.h, wmi.c,wmiconfig.c to pass compile.
4. design directory structure to make adding new other vendor wifi clearly.
5. add CleanSpec.mk to clean middle compile file when do clean.
6. add .gitignore to ignore some tained files.
7. add auto generate softmac feature.check_wifi_mac.sh can generate a
   different mac for different board in /device/wifi/softmac.

Change-Id: I9573e4f4fbf2afbaf6763254bc18ae37552b7533
Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 .gitignore                                         |  12 +
 Android.mk                                         |   2 +-
 wlan/Android.mk                                    |   4 +
 wlan/CleanSpec.mk                                  |  53 ++
 wlan/atheros/Android.mk                            |   2 +
 wlan/atheros/compat-wireless/Android.mk            |  67 ++
 wlan/atheros/compat-wireless/check_wifi_mac.sh     |  13 +
 .../drivers/net/wireless/ath/ath6kl/platform.c     |   1 -
 .../drivers/net/wireless/ath/ath6kl/softmac.c      |  71 +-
 .../drivers/net/wireless/ath/ath6kl/wmi.c          |   3 +
 .../drivers/net/wireless/ath/ath6kl/wmiconfig.c    |   3 +-
 .../include/linux/compat_autoconf.h                | 791 ++++++++++++++++++++-
 .../atheros/compat-wireless/include/linux/export.h |  20 +-
 13 files changed, 980 insertions(+), 62 deletions(-)
 create mode 100644 .gitignore
 create mode 100644 wlan/Android.mk
 create mode 100644 wlan/CleanSpec.mk
 create mode 100644 wlan/atheros/Android.mk
 create mode 100644 wlan/atheros/compat-wireless/Android.mk
 create mode 100755 wlan/atheros/compat-wireless/check_wifi_mac.sh
 mode change 100644 => 100755 wlan/atheros/compat-wireless/scripts/check_config.sh
 mode change 100644 => 100755 wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..99cfdb5
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,12 @@
+.tmp_versions
+modules.order
+modules
+Module.symvers
+*.o.cmd
+*.o
+*.ko.cmd
+*.ko
+.compat_autoconf_patch*
+.config.mk_md5sum.txt
+*.mod.c
+regdb.c*
diff --git a/Android.mk b/Android.mk
index ea6c5a5..1ec56a7 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,4 +1,4 @@
-common_imx_dirs := libsensors alsa libgps lights
+common_imx_dirs := libsensors alsa libgps lights wlan
 mx5x_dirs := $(common_imx_dirs) mx5x/libcopybit mx5x/libgralloc  mx5x/hwcomposer mx5x/libcamera
 mx6_dirs := $(common_imx_dirs) mx6/libgralloc_wrapper mx6/hwcomposer mx6/libcamera
 
diff --git a/wlan/Android.mk b/wlan/Android.mk
new file mode 100644
index 0000000..1cb4fac
--- /dev/null
+++ b/wlan/Android.mk
@@ -0,0 +1,4 @@
+atheros_dirs := compat-wireless
+ifeq ($(BOARD_HAS_ATH_WLAN),true)
+    include $(call all-subdir-makefiles,$(atheros_dirs))
+endif
diff --git a/wlan/CleanSpec.mk b/wlan/CleanSpec.mk
new file mode 100644
index 0000000..19cf276
--- /dev/null
+++ b/wlan/CleanSpec.mk
@@ -0,0 +1,53 @@
+# Copyright (C) 2007 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# If you don't need to do a full clean build but would like to touch
+# a file or delete some intermediate files, add a clean step to the end
+# of the list.  These steps will only be run once, if they haven't been
+# run before.
+#
+# E.g.:
+#     $(call add-clean-step, touch -c external/sqlite/sqlite3.h)
+#     $(call add-clean-step, rm -rf $(PRODUCT_OUT)/obj/STATIC_LIBRARIES/libz_intermediates)
+#
+# Always use "touch -c" and "rm -f" or "rm -rf" to gracefully deal with
+# files that are missing or have been moved.
+#
+# Use $(PRODUCT_OUT) to get to the "out/target/product/blah/" directory.
+# Use $(OUT_DIR) to refer to the "out" directory.
+#
+# If you need to re-do something that's already mentioned, just copy
+# the command and add it to the bottom of the list.  E.g., if a change
+# that you made last week required touching a file and a change you
+# made today requires touching the same file, just copy the old
+# touch step and add it to the end of the list.
+#
+# ************************************************
+# NEWER CLEAN STEPS MUST BE AT THE END OF THE LIST
+# ************************************************
+
+# For example:
+#$(call add-clean-step, rm -rf $(OUT_DIR)/target/common/obj/APPS/AndroidTests_intermediates)
+#$(call add-clean-step, rm -rf $(OUT_DIR)/target/common/obj/JAVA_LIBRARIES/core_intermediates)
+#$(call add-clean-step, find $(OUT_DIR) -type f -name "IGTalkSession*" -print0 | xargs -0 rm -f)
+#$(call add-clean-step, rm -rf $(PRODUCT_OUT)/data/*)
+
+# ************************************************
+# NEWER CLEAN STEPS MUST BE AT THE END OF THE LIST
+# ************************************************
+$(call add-clean-step, rm -rf hardware/imx/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/ath6kl_sdio.ko)
+$(call add-clean-step, rm -rf hardware/imx/wlan/atheros/compat-wireless/net/wireless/cfg80211.ko)
+$(call add-clean-step, rm -rf hardware/imx/wlan/atheros/compat-wireless/compat/compat.ko)
+
diff --git a/wlan/atheros/Android.mk b/wlan/atheros/Android.mk
new file mode 100644
index 0000000..8338432
--- /dev/null
+++ b/wlan/atheros/Android.mk
@@ -0,0 +1,2 @@
+include $(call all-subdir-makefiles)
+
diff --git a/wlan/atheros/compat-wireless/Android.mk b/wlan/atheros/compat-wireless/Android.mk
new file mode 100644
index 0000000..fb8be9e
--- /dev/null
+++ b/wlan/atheros/compat-wireless/Android.mk
@@ -0,0 +1,67 @@
+#------------------------------------------------------------------------------
+# <copyright file="makefile" company="Atheros">
+# Copyright (c) 2005-2010 Atheros Corporation.  All rights reserved.
+# Copyright (c)2010-2012 Freescale Semiconductor, Inc.
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+#
+#------------------------------------------------------------------------------
+#==============================================================================
+# Author(s): ="Atheros"
+#==============================================================================
+
+ifneq ($(TARGET_SIMULATOR),true)
+ifeq ($(BOARD_WLAN_DEVICE),ar6003)
+
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+ATH_ANDROID_SRC_BASE:= $(LOCAL_PATH)
+
+ATH_ANDROID_ROOT:= $(CURDIR)
+
+ATH_LINUXPATH=$(ATH_ANDROID_ROOT)/kernel_imx
+
+mod_cleanup := $(ATH_ANDROID_ROOT)/$(ATH_ANDROID_SRC_BASE)/dummy
+
+$(mod_cleanup) :
+	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) ARCH=arm CROSS_COMPILE=$(ARM_EABI_TOOLCHAIN)/arm-eabi- KLIB=$(AT\
+  H_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH) clean
+	mkdir -p $(TARGET_OUT)/etc/firmware/ath6k/AR6003/hw2.1.1/
+	mkdir -p $(TARGET_OUT)/lib/modules/
+
+ath6kl_module_file :=drivers/net/wireless/ath/ath6kl/ath6kl_sdio.ko
+$(ATH_ANDROID_SRC_BASE)/$(ath6kl_module_file):$(mod_cleanup) $(TARGET_PREBUILT_KERNEL) $(ACP)
+	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) O=$(ATH_LINUXPATH) ARCH=arm CROSS_COMPILE=$(ARM_EABI_TOOLCHAIN)/arm-eabi- KLIB=$(ATH_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH)
+	$(ACP) -fpt $(ATH_ANDROID_SRC_BASE)/compat/compat.ko $(TARGET_OUT)/lib/modules/
+	$(ACP) -fpt $(ATH_ANDROID_SRC_BASE)/net/wireless/cfg80211.ko $(TARGET_OUT)/lib/modules/
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := ath6kl_sdio.ko
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT)/lib/modules
+LOCAL_SRC_FILES := $(ath6kl_module_file)
+include $(BUILD_PREBUILT)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := check_wifi_mac.sh
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := EXECUTABLES
+LOCAL_MODULE_PATH := $(TARGET_OUT)/etc
+LOCAL_SRC_FILES := check_wifi_mac.sh
+include $(BUILD_PREBUILT)
+
+endif
+endif
diff --git a/wlan/atheros/compat-wireless/check_wifi_mac.sh b/wlan/atheros/compat-wireless/check_wifi_mac.sh
new file mode 100755
index 0000000..7b6e932
--- /dev/null
+++ b/wlan/atheros/compat-wireless/check_wifi_mac.sh
@@ -0,0 +1,13 @@
+if [ -e /device/wifi/softmac ]
+then
+	echo "update softmac"
+else
+	mount -w -o remount /device
+	mkdir /device/wifi
+	touch /device/wifi/softmac
+	echo 02:30:`busybox expr $RANDOM % 89 + 10`:`busybox expr $RANDOM % 89 + 10`:`busybox expr $RANDOM % 89 + 10`:`busybox expr $RANDOM % 89 + 10` > /device/wifi/softmac
+	sync
+	chmod 775 /vendor/wifi
+	chmod 664 /vendor/wifi/softmac
+	mount -r -o remount /device
+fi
diff --git a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/platform.c b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/platform.c
index f91ff5f..d31a170 100644
--- a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/platform.c
+++ b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/platform.c
@@ -84,7 +84,6 @@ int android_readwrite_file(const char *filename,
 		filp_close(filp, NULL);
 
 	set_fs(oldfs);
-	printk(KERN_ERR "android_readwrite_file: ret=%d\n", ret);
 
 	return ret;
 }
diff --git a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c
index a08891b..41f08b7 100644
--- a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c
+++ b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c
@@ -1,6 +1,8 @@
 /*
  * Copyright (c) 2011 Atheros Communications Inc.
  *
+ * Copyright (c) 2010-2012 Freescale Semiconductor, Inc.
+ *
  * Permission to use, copy, modify, and/or distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
  * copyright notice and this permission notice appear in all copies.
@@ -18,6 +20,7 @@
 #include "debug.h"
 #include <linux/vmalloc.h>
 #define MAC_FILE "ath6k/AR6003/hw2.1.1/softmac"
+#define VENDOR_MAC_FILE "/device/wifi/softmac"
 
 typedef char            A_CHAR;
 extern int android_readwrite_file(const A_CHAR *filename, A_CHAR *rbuf, const A_CHAR *wbuf, size_t length);
@@ -29,6 +32,8 @@ extern int android_readwrite_file(const A_CHAR *filename, A_CHAR *rbuf, const A_
 /* Global variables, sane coding be damned. */
 u8 *ath6kl_softmac;
 size_t ath6kl_softmac_len;
+/*0=from vendor 1=from firmware*/
+int macsource = 0;
 
 static void ath6kl_calculate_crc(u32 target_type, u8 *data, size_t len)
 {
@@ -62,18 +67,14 @@ static void ath6kl_calculate_crc(u32 target_type, u8 *data, size_t len)
 	ath6kl_dbg(ATH6KL_DBG_BOOT, "New Checksum: %u\n", checksum);
 }
 
-#ifdef CONFIG_MACH_PX
-static int ath6kl_fetch_nvmac_info(struct ath6kl *ar)
+static int ath6kl_fetch_mac_from_file(struct ath6kl *ar, u8* filename)
 {
 	char softmac_filename[256];
 	int ret = 0;
 
 	do {
-		snprintf(softmac_filename, sizeof(softmac_filename),
-			 "/data/.nvmac.info");
-
 		if ( (ret = android_readwrite_file(
-			softmac_filename, NULL, NULL, 0)) < 0) {
+			filename, NULL, NULL, 0)) < 0) {
 			break;
 		} else {
 			ath6kl_softmac_len = ret;
@@ -87,7 +88,7 @@ static int ath6kl_fetch_nvmac_info(struct ath6kl *ar)
 			break;
 		}
 
-		if ( (ret = android_readwrite_file(softmac_filename,
+		if ( (ret = android_readwrite_file(filename,
 			 (char*)ath6kl_softmac, NULL, ath6kl_softmac_len)) !=
 				 ath6kl_softmac_len) {
 			ath6kl_dbg(ATH6KL_DBG_BOOT,"%s: file read error, length %d\n",
@@ -96,7 +97,7 @@ static int ath6kl_fetch_nvmac_info(struct ath6kl *ar)
 			ret = -1;
 			break;
 		}
-
+		ath6kl_dbg(ATH6KL_DBG_BOOT,"Get mac from file =%s\r\n",ath6kl_softmac);
 		if (!strncmp(ath6kl_softmac,"00:00:00:00:00:00",17)) {
 			ath6kl_dbg(ATH6KL_DBG_BOOT,"%s: mac address is zero\n",
 					 __func__);
@@ -107,17 +108,15 @@ static int ath6kl_fetch_nvmac_info(struct ath6kl *ar)
 
 		ret = 0;
 	} while (0);
-
+	macsource = 0;
 	return ret;
 }
 
-#else
-static int ath6kl_fetch_mac_file(struct ath6kl *ar)
+static int ath6kl_fetch_mac_from_firmware(struct ath6kl *ar)
 {
 	const struct firmware *fw_entry;
 	int ret = 0;
 
-
 	ret = request_firmware(&fw_entry, MAC_FILE, ar->dev);
 	if (ret)
 		return ret;
@@ -129,6 +128,23 @@ static int ath6kl_fetch_mac_file(struct ath6kl *ar)
 		ret = -ENOMEM;
 
 	release_firmware(fw_entry);
+	macsource = 1;
+	return ret;
+}
+#ifdef CONFIG_MACH_PX
+static int ath6kl_fetch_nvmac_info(struct ath6kl *ar)
+{
+	return ath6kl_fetch_mac_from_file(ar, "/data/.nvmac.info");
+}
+
+#else
+static int ath6kl_fetch_mac_file(struct ath6kl *ar)
+{
+	int ret = 0;
+	ret = ath6kl_fetch_mac_from_file(ar, VENDOR_MAC_FILE);
+
+	if ( ret < 0)
+		return ath6kl_fetch_mac_from_firmware(ar);
 
 	return ret;
 }
@@ -138,9 +154,9 @@ void ath6kl_mangle_mac_address(struct ath6kl *ar, u8 locally_administered_bit)
 {
 	u8 *ptr_mac;
 	int i, ret;
-#ifdef CONFIG_MACH_PX
+
 	unsigned int softmac[6];
-#endif
+
 
 	switch (ar->target_type) {
 	case TARGET_TYPE_AR6003:
@@ -169,13 +185,10 @@ void ath6kl_mangle_mac_address(struct ath6kl *ar, u8 locally_administered_bit)
 
 	if (sscanf(ath6kl_softmac, "%02x:%02x:%02x:%02x:%02x:%02x",
 				&softmac[0], &softmac[1], &softmac[2],
-				&softmac[3], &softmac[4], &softmac[5])==6) {
-
-		for (i=0; i<6; ++i) {
-			ptr_mac[i] = softmac[i] & 0xff;
-		}
+		                &softmac[3], &softmac[4], &softmac[5])==6) {
+		for (i=0; i<6; ++i)
+		     ptr_mac[i] = softmac[i] & 0xff;
 	}
-
 	ath6kl_dbg(ATH6KL_DBG_BOOT,
 			"MAC from SoftMAC %02X:%02X:%02X:%02X:%02X:%02X\n",
 			ptr_mac[0], ptr_mac[1], ptr_mac[2],
@@ -187,16 +200,22 @@ void ath6kl_mangle_mac_address(struct ath6kl *ar, u8 locally_administered_bit)
 		ath6kl_err("MAC address file not found\n");
 		return;
 	}
-
-	for (i = 0; i < ETH_ALEN; ++i) {
-	   ptr_mac[i] = ath6kl_softmac[i] & 0xff;
+	if ( macsource == 0) {
+		if (sscanf(ath6kl_softmac, "%02x:%02x:%02x:%02x:%02x:%02x",
+				&softmac[0], &softmac[1], &softmac[2],
+				   &softmac[3], &softmac[4], &softmac[5])==6 ) {
+			for (i=0; i<6; ++i)
+				ptr_mac[i] = softmac[i] & 0xff;
+		}
+		vfree(ath6kl_softmac);
+	} else if (macsource == 1) {
+		for (i = 0; i < ETH_ALEN; ++i)
+		    ptr_mac[i] = ath6kl_softmac[i] & 0xff;
+		kfree(ath6kl_softmac);
 	}
 
-	kfree(ath6kl_softmac);
 #endif
-
 	if (locally_administered_bit)
 		ptr_mac[0] |= 0x02;
-
 	ath6kl_calculate_crc(ar->target_type, ar->fw_board, ar->fw_board_len);
 }
diff --git a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmi.c b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmi.c
index e705a7b..4200d69 100644
--- a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmi.c
+++ b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmi.c
@@ -1,6 +1,7 @@
 /*
  * Copyright (c) 2004-2011 Atheros Communications Inc.
  * Copyright (c) 2011-2012 Qualcomm Atheros, Inc.
+ * Copyright (c) 2010-2012 Freescale Semiconductor, Inc.
  *
  * Permission to use, copy, modify, and/or distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -3765,7 +3766,9 @@ int ath6kl_wmi_control_rx(struct wmi *wmi, struct sk_buff *skb)
 		break;
 	case WMI_ERROR_REPORT_EVENTID:
 		ath6kl_dbg(ATH6KL_DBG_WMI, "WMI_ERROR_REPORT_EVENTID\n");
+#ifdef CONFIG_ATH6KL_DEBUG
 		ath6kl_wmi_error_report_event(wmi, datap, len);
+#endif
 		cfg80211_send_event_to_app(skb->dev, id, datap, len);
 		break;
 	case WMI_OPT_RX_FRAME_EVENTID:
diff --git a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmiconfig.c b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmiconfig.c
index bb60ed2..88763c1 100644
--- a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmiconfig.c
+++ b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/wmiconfig.c
@@ -1,6 +1,7 @@
 /*
  * Copyright (c) 2010-2011 Atheros Communications Inc.
  * Copyright (c) 2011 Qualcomm Atheros, Inc.
+ * Copyright (c) 2010-2012 Freescale Semiconductor, Inc.
  *
  * Permission to use, copy, modify, and/or distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -18,7 +19,7 @@
 #include "testmode.h"
 #include "debug.h"
 #include "wmi.h"
-
+#include "cfg80211.h"
 #include <net/netlink.h>
 
 enum ath6kl_tm_attr {
diff --git a/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h b/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
index 813755c..a7d6487 100644
--- a/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
+++ b/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
@@ -1,37 +1,764 @@
 #ifndef COMPAT_AUTOCONF_INCLUDED
 #define COMPAT_AUTOCONF_INCLUDED
-
 /*
- * This is an example of a compat autconf header file that gets
- * automatically generated by your projet. You will only need this
- * if you are working with a lot of config options.
- *
- * Your project will likely just override this, or leave this as-is.
+ * Automatically generated C config: don't edit
+ * Sat Sep 29 15:20:31 CST 2012 
+ * compat-wireless-2.6: patch-10-22-gc53d81a-p
+ * linux-2.6: msm-2012-06-21-43-g7ffaacb
  */
-
-/*
- * Example automatically generated C config: don't edit
- * Thu Dec 10 10:34:11 PST 2009 
- * compat-wireless-2.6: master-2009-11-19-5-gb4fd4dd
- * linux-2.6: next-20091208
- */
-//#define COMPAT_PROJECT_FOO_RELEASE "master-2009-11-19-5-gb4fd4dd"
-//#define COMPAT_UPSTREAM_FOO_KERNEL_RELEASE "next-20091208"
-
-/* Example kernel version minimum requirement */
-//#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,10))
-//#error Compat-wireless requirement: Linux >= 2,6,10
-//#endif /* (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,10) */ 
-
-/* Example hard requirement */
-//#error Compat-wireless requirement: CONFIG_WIRELESS_EXT must be enabled in your kernel
-//#endif /* CONFIG_WIRELESS_EXT */
-
-/*
- * Example of how to match your external modules kconfig options into this.
- * You'll need something to generate this for you.
- */
-//#ifndef CONFIG_FOO_BAR
-//#define CONFIG_FOO_BAR 1
-
+#define COMPAT_RELEASE "patch-10-22-gc53d81a-p"
+#define COMPAT_KERNEL_RELEASE "msm-2012-06-21-43-g7ffaacb"
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
+#error Compat-wireless requirement: Linux >= 2,6,24
+#endif /* (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24) */ 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
+#endif /* CONFIG_COMPAT_KERNEL_2_6_24 */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,27))
+#endif /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,33))
+#if (!defined(RHEL_MAJOR) || RHEL_MAJOR != 6 || RHEL_MINOR < 1)
+#if defined(CONFIG_FW_LOADER) || defined(CONFIG_FW_LOADER_MODULE)
+#ifndef CONFIG_COMPAT_FIRMWARE_CLASS
+#define CONFIG_COMPAT_FIRMWARE_CLASS 1
+#endif /* CONFIG_COMPAT_FIRMWARE_CLASS */ 
+#endif /* CONFIG_FW_LOADER */
+#endif /* CONFIG_COMPAT_RHEL_6_1 */
+#endif /* CONFIG_COMPAT_KERNEL_2_6_33 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,36))
+#if (!defined(RHEL_MAJOR) || RHEL_MAJOR != 6 || RHEL_MINOR < 1)
+#endif /* CONFIG_COMPAT_RHEL_6_1 */
+#endif /* CONFIG_COMPAT_KERNEL_2_6_36 */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,33))
+#ifndef CONFIG_COMPAT_BT_SOCK_CREATE_NEEDS_KERN
+#define CONFIG_COMPAT_BT_SOCK_CREATE_NEEDS_KERN 1
+#endif /* CONFIG_COMPAT_BT_SOCK_CREATE_NEEDS_KERN */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_33 */
+#if (defined(RHEL_MAJOR) && RHEL_MAJOR == 6 && RHEL_MINOR >= 0)
+#ifndef CONFIG_COMPAT_BT_SOCK_CREATE_NEEDS_KERN
+#define CONFIG_COMPAT_BT_SOCK_CREATE_NEEDS_KERN 1
+#endif /* CONFIG_COMPAT_BT_SOCK_CREATE_NEEDS_KERN */ 
+#endif /* CONFIG_COMPAT_RHEL_6_0 */
+#if (defined(RHEL_MAJOR) && RHEL_MAJOR == 6 && RHEL_MINOR >= 0)
+#ifndef CONFIG_COMPAT_FIRMWARE_DATA_RW_NEEDS_FILP
+#define CONFIG_COMPAT_FIRMWARE_DATA_RW_NEEDS_FILP 1
+#endif /* CONFIG_COMPAT_FIRMWARE_DATA_RW_NEEDS_FILP */ 
+#endif /* CONFIG_COMPAT_RHEL_6_0 */
+#ifndef CONFIG_MAC80211
+#define CONFIG_MAC80211 1
+#endif /* CONFIG_MAC80211 */ 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,33))
+#ifndef CONFIG_MAC80211_DRIVER_API_TRACER
+#define CONFIG_MAC80211_DRIVER_API_TRACER 1
+#endif /* CONFIG_MAC80211_DRIVER_API_TRACER */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_33 */
+#ifndef CONFIG_MAC80211_RC_DEFAULT
+#define CONFIG_MAC80211_RC_DEFAULT "minstrel_ht"
+#endif /* CONFIG_MAC80211_RC_DEFAULT */ 
+#ifndef CONFIG_MAC80211_RC_DEFAULT_MINSTREL
+#define CONFIG_MAC80211_RC_DEFAULT_MINSTREL 1
+#endif /* CONFIG_MAC80211_RC_DEFAULT_MINSTREL */ 
+#ifndef CONFIG_COMPAT_MAC80211_RC_DEFAULT
+#define CONFIG_COMPAT_MAC80211_RC_DEFAULT "minstrel_ht"
+#endif /* CONFIG_COMPAT_MAC80211_RC_DEFAULT */ 
+#ifndef CONFIG_MAC80211_RC_PID
+#define CONFIG_MAC80211_RC_PID 1
+#endif /* CONFIG_MAC80211_RC_PID */ 
+#ifndef CONFIG_MAC80211_RC_MINSTREL
+#define CONFIG_MAC80211_RC_MINSTREL 1
+#endif /* CONFIG_MAC80211_RC_MINSTREL */ 
+#ifndef CONFIG_MAC80211_RC_MINSTREL_HT
+#define CONFIG_MAC80211_RC_MINSTREL_HT 1
+#endif /* CONFIG_MAC80211_RC_MINSTREL_HT */ 
+#if defined(CONFIG_LEDS_TRIGGERS) || defined(CONFIG_LEDS_TRIGGERS_MODULE)
+#ifndef CONFIG_MAC80211_LEDS
+#define CONFIG_MAC80211_LEDS 1
+#endif /* CONFIG_MAC80211_LEDS */ 
+#endif /* CONFIG_LEDS_TRIGGERS */
+#ifndef CONFIG_MAC80211_MESH
+#define CONFIG_MAC80211_MESH 1
+#endif /* CONFIG_MAC80211_MESH */ 
+#ifndef CONFIG_CFG80211
+#define CONFIG_CFG80211 1
+#endif /* CONFIG_CFG80211 */ 
+#ifndef CONFIG_CFG80211_DEFAULT_PS
+#define CONFIG_CFG80211_DEFAULT_PS 1
+#endif /* CONFIG_CFG80211_DEFAULT_PS */ 
+#ifndef CONFIG_NL80211_TESTMODE
+#define CONFIG_NL80211_TESTMODE 1
+#endif /* CONFIG_NL80211_TESTMODE */ 
+#ifndef CONFIG_CFG80211_INTERNAL_REGDB
+#define CONFIG_CFG80211_INTERNAL_REGDB 1
+#endif /* CONFIG_CFG80211_INTERNAL_REGDB */ 
+#ifndef CONFIG_LIB80211
+#define CONFIG_LIB80211 1
+#endif /* CONFIG_LIB80211 */ 
+#ifndef CONFIG_LIB80211_CRYPT_WEP
+#define CONFIG_LIB80211_CRYPT_WEP 1
+#endif /* CONFIG_LIB80211_CRYPT_WEP */ 
+#ifndef CONFIG_LIB80211_CRYPT_CCMP
+#define CONFIG_LIB80211_CRYPT_CCMP 1
+#endif /* CONFIG_LIB80211_CRYPT_CCMP */ 
+#ifndef CONFIG_LIB80211_CRYPT_TKIP
+#define CONFIG_LIB80211_CRYPT_TKIP 1
+#endif /* CONFIG_LIB80211_CRYPT_TKIP */ 
+#ifndef CONFIG_BT
+#define CONFIG_BT 1
+#endif /* CONFIG_BT */ 
+#ifndef CONFIG_COMPAT_BT_L2CAP
+#define CONFIG_COMPAT_BT_L2CAP 1
+#endif /* CONFIG_COMPAT_BT_L2CAP */ 
+#ifndef CONFIG_COMPAT_BT_SCO
+#define CONFIG_COMPAT_BT_SCO 1
+#endif /* CONFIG_COMPAT_BT_SCO */ 
+#ifndef CONFIG_BT_RFCOMM
+#define CONFIG_BT_RFCOMM 1
+#endif /* CONFIG_BT_RFCOMM */ 
+#ifndef CONFIG_BT_RFCOMM_TTY
+#define CONFIG_BT_RFCOMM_TTY 1
+#endif /* CONFIG_BT_RFCOMM_TTY */ 
+#ifndef CONFIG_BT_BNEP
+#define CONFIG_BT_BNEP 1
+#endif /* CONFIG_BT_BNEP */ 
+#ifndef CONFIG_BT_BNEP_MC_FILTER
+#define CONFIG_BT_BNEP_MC_FILTER 1
+#endif /* CONFIG_BT_BNEP_MC_FILTER */ 
+#ifndef CONFIG_BT_BNEP_PROTO_FILTER
+#define CONFIG_BT_BNEP_PROTO_FILTER 1
+#endif /* CONFIG_BT_BNEP_PROTO_FILTER */ 
+#if defined(CONFIG_ISDN_CAPI) || defined(CONFIG_ISDN_CAPI_MODULE)
+#ifndef CONFIG_BT_CMTP
+#define CONFIG_BT_CMTP 1
+#endif /* CONFIG_BT_CMTP */ 
+#endif /* CONFIG_ISDN_CAPI */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,28))
+#ifndef CONFIG_COMPAT_BT_HIDP
+#define CONFIG_COMPAT_BT_HIDP 1
+#endif /* CONFIG_COMPAT_BT_HIDP */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_28 */
+#ifndef CONFIG_BT_HCIUART
+#define CONFIG_BT_HCIUART "M"
+#endif /* CONFIG_BT_HCIUART */ 
+#ifndef CONFIG_BT_HCIUART_H4
+#define CONFIG_BT_HCIUART_H4 1
+#endif /* CONFIG_BT_HCIUART_H4 */ 
+#ifndef CONFIG_BT_HCIUART_BCSP
+#define CONFIG_BT_HCIUART_BCSP 1
+#endif /* CONFIG_BT_HCIUART_BCSP */ 
+#ifndef CONFIG_BT_HCIUART_ATH3K
+#define CONFIG_BT_HCIUART_ATH3K 1
+#endif /* CONFIG_BT_HCIUART_ATH3K */ 
+#ifndef CONFIG_BT_HCIUART_LL
+#define CONFIG_BT_HCIUART_LL 1
+#endif /* CONFIG_BT_HCIUART_LL */ 
+#ifndef CONFIG_BT_HCIVHCI
+#define CONFIG_BT_HCIVHCI 1
+#endif /* CONFIG_BT_HCIVHCI */ 
+#ifndef CONFIG_BT_MRVL
+#define CONFIG_BT_MRVL 1
+#endif /* CONFIG_BT_MRVL */ 
+#if defined(CONFIG_PCMCIA) || defined(CONFIG_PCMCIA_MODULE)
+#ifndef CONFIG_BT_HCIDTL1
+#define CONFIG_BT_HCIDTL1 1
+#endif /* CONFIG_BT_HCIDTL1 */ 
+#ifndef CONFIG_BT_HCIBT3C
+#define CONFIG_BT_HCIBT3C 1
+#endif /* CONFIG_BT_HCIBT3C */ 
+#ifndef CONFIG_BT_HCIBLUECARD
+#define CONFIG_BT_HCIBLUECARD 1
+#endif /* CONFIG_BT_HCIBLUECARD */ 
+#ifndef CONFIG_BT_HCIBTUART
+#define CONFIG_BT_HCIBTUART 1
+#endif /* CONFIG_BT_HCIBTUART */ 
+#endif /* CONFIG_PCMCIA */
+#if defined(CONFIG_WIRELESS_EXT) || defined(CONFIG_WIRELESS_EXT_MODULE)
+#ifndef CONFIG_CFG80211_WEXT
+#define CONFIG_CFG80211_WEXT 1
+#endif /* CONFIG_CFG80211_WEXT */ 
+#else /* CONFIG_CFG80211_WEXT */
+#endif /* CONFIG_WIRELESS_EXT */
+#if defined(CONFIG_STAGING) || defined(CONFIG_STAGING_MODULE)
+#ifndef CONFIG_COMPAT_STAGING
+#define CONFIG_COMPAT_STAGING 1
+#endif /* CONFIG_COMPAT_STAGING */ 
+#endif /* CONFIG_STAGING */
+#ifndef CONFIG_MAC80211_HWSIM
+#define CONFIG_MAC80211_HWSIM 1
+#endif /* CONFIG_MAC80211_HWSIM */ 
+#ifndef CONFIG_ATH5K
+#define CONFIG_ATH5K 1
+#endif /* CONFIG_ATH5K */ 
+#ifndef CONFIG_ATH9K
+#define CONFIG_ATH9K 1
+#endif /* CONFIG_ATH9K */ 
+#ifndef CONFIG_ATH9K_HW
+#define CONFIG_ATH9K_HW 1
+#endif /* CONFIG_ATH9K_HW */ 
+#ifndef CONFIG_ATH9K_COMMON
+#define CONFIG_ATH9K_COMMON 1
+#endif /* CONFIG_ATH9K_COMMON */ 
+#ifndef CONFIG_ATH9K_RATE_CONTROL
+#define CONFIG_ATH9K_RATE_CONTROL 1
+#endif /* CONFIG_ATH9K_RATE_CONTROL */ 
+#if defined(CONFIG_PCI) || defined(CONFIG_PCI_MODULE)
+#ifndef CONFIG_ATH5K_PCI
+#define CONFIG_ATH5K_PCI 1
+#endif /* CONFIG_ATH5K_PCI */ 
+#ifndef CONFIG_ATH9K_PCI
+#define CONFIG_ATH9K_PCI 1
+#endif /* CONFIG_ATH9K_PCI */ 
+#ifndef CONFIG_IWLWIFI
+#define CONFIG_IWLWIFI 1
+#endif /* CONFIG_IWLWIFI */ 
+#ifndef CONFIG_IWLEGACY
+#define CONFIG_IWLEGACY 1
+#endif /* CONFIG_IWLEGACY */ 
+#ifndef CONFIG_COMPAT_IWL4965
+#define CONFIG_COMPAT_IWL4965 1
+#endif /* CONFIG_COMPAT_IWL4965 */ 
+#ifndef CONFIG_IWL3945
+#define CONFIG_IWL3945 1
+#endif /* CONFIG_IWL3945 */ 
+#ifndef CONFIG_B43
+#define CONFIG_B43 1
+#endif /* CONFIG_B43 */ 
+#ifndef CONFIG_B43_HWRNG
+#define CONFIG_B43_HWRNG 1
+#endif /* CONFIG_B43_HWRNG */ 
+#ifndef CONFIG_B43_PCI_AUTOSELECT
+#define CONFIG_B43_PCI_AUTOSELECT 1
+#endif /* CONFIG_B43_PCI_AUTOSELECT */ 
+#if defined(CONFIG_PCMCIA) || defined(CONFIG_PCMCIA_MODULE)
+#ifndef CONFIG_B43_PCMCIA
+#define CONFIG_B43_PCMCIA 1
+#endif /* CONFIG_B43_PCMCIA */ 
+#endif /* CONFIG_PCMCIA */
+#if defined(CONFIG_MAC80211_LEDS) || defined(CONFIG_MAC80211_LEDS_MODULE)
+#ifndef CONFIG_B43_LEDS
+#define CONFIG_B43_LEDS 1
+#endif /* CONFIG_B43_LEDS */ 
+#endif /* CONFIG_MAC80211_LEDS */
+#ifndef CONFIG_B43_PHY_LP
+#define CONFIG_B43_PHY_LP 1
+#endif /* CONFIG_B43_PHY_LP */ 
+#ifndef CONFIG_B43_PHY_N
+#define CONFIG_B43_PHY_N 1
+#endif /* CONFIG_B43_PHY_N */ 
+#ifndef CONFIG_B43_PHY_HT
+#define CONFIG_B43_PHY_HT 1
+#endif /* CONFIG_B43_PHY_HT */ 
+#ifndef CONFIG_B43LEGACY
+#define CONFIG_B43LEGACY 1
+#endif /* CONFIG_B43LEGACY */ 
+#ifndef CONFIG_B43LEGACY_HWRNG
+#define CONFIG_B43LEGACY_HWRNG 1
+#endif /* CONFIG_B43LEGACY_HWRNG */ 
+#ifndef CONFIG_B43LEGACY_PCI_AUTOSELECT
+#define CONFIG_B43LEGACY_PCI_AUTOSELECT 1
+#endif /* CONFIG_B43LEGACY_PCI_AUTOSELECT */ 
+#if defined(CONFIG_MAC80211_LEDS) || defined(CONFIG_MAC80211_LEDS_MODULE)
+#ifndef CONFIG_B43LEGACY_LEDS
+#define CONFIG_B43LEGACY_LEDS 1
+#endif /* CONFIG_B43LEGACY_LEDS */ 
+#endif /* CONFIG_MAC80211_LEDS */
+#ifndef CONFIG_B43LEGACY_DMA
+#define CONFIG_B43LEGACY_DMA 1
+#endif /* CONFIG_B43LEGACY_DMA */ 
+#ifndef CONFIG_B43LEGACY_PIO
+#define CONFIG_B43LEGACY_PIO 1
+#endif /* CONFIG_B43LEGACY_PIO */ 
+#if defined(CONFIG_WIRELESS_EXT) || defined(CONFIG_WIRELESS_EXT_MODULE)
+#ifndef CONFIG_LIBIPW
+#define CONFIG_LIBIPW 1
+#endif /* CONFIG_LIBIPW */ 
+#ifndef CONFIG_IPW2100
+#define CONFIG_IPW2100 1
+#endif /* CONFIG_IPW2100 */ 
+#ifndef CONFIG_IPW2100_MONITOR
+#define CONFIG_IPW2100_MONITOR 1
+#endif /* CONFIG_IPW2100_MONITOR */ 
+#ifndef CONFIG_IPW2200
+#define CONFIG_IPW2200 1
+#endif /* CONFIG_IPW2200 */ 
+#ifndef CONFIG_IPW2200_MONITOR
+#define CONFIG_IPW2200_MONITOR 1
+#endif /* CONFIG_IPW2200_MONITOR */ 
+#ifndef CONFIG_IPW2200_RADIOTAP
+#define CONFIG_IPW2200_RADIOTAP 1
+#endif /* CONFIG_IPW2200_RADIOTAP */ 
+#ifndef CONFIG_IPW2200_PROMISCUOUS
+#define CONFIG_IPW2200_PROMISCUOUS 1
+#endif /* CONFIG_IPW2200_PROMISCUOUS */ 
+#ifndef CONFIG_IPW2200_QOS
+#define CONFIG_IPW2200_QOS 1
+#endif /* CONFIG_IPW2200_QOS */ 
+#endif /* CONFIG_WIRELESS_EXT */
+#if defined(CONFIG_SSB) || defined(CONFIG_SSB_MODULE)
+#ifndef CONFIG_SSB_SPROM
+#define CONFIG_SSB_SPROM 1
+#endif /* CONFIG_SSB_SPROM */ 
+#ifndef CONFIG_SSB_BLOCKIO
+#define CONFIG_SSB_BLOCKIO 1
+#endif /* CONFIG_SSB_BLOCKIO */ 
+#ifndef CONFIG_SSB_PCIHOST
+#define CONFIG_SSB_PCIHOST 1
+#endif /* CONFIG_SSB_PCIHOST */ 
+#ifndef CONFIG_SSB_B43_PCI_BRIDGE
+#define CONFIG_SSB_B43_PCI_BRIDGE 1
+#endif /* CONFIG_SSB_B43_PCI_BRIDGE */ 
+#if defined(CONFIG_PCMCIA) || defined(CONFIG_PCMCIA_MODULE)
+#ifndef CONFIG_SSB_PCMCIAHOST
+#define CONFIG_SSB_PCMCIAHOST 1
+#endif /* CONFIG_SSB_PCMCIAHOST */ 
+#endif /* CONFIG_PCMCIA */
+#ifndef CONFIG_SSB_DRIVER_PCICORE
+#define CONFIG_SSB_DRIVER_PCICORE 1
+#endif /* CONFIG_SSB_DRIVER_PCICORE */ 
+#ifndef CONFIG_B43_SSB
+#define CONFIG_B43_SSB 1
+#endif /* CONFIG_B43_SSB */ 
+#endif /* CONFIG_SSB */
+#ifndef CONFIG_BCMA
+#define CONFIG_BCMA 1
+#endif /* CONFIG_BCMA */ 
+#ifndef CONFIG_BCMA_BLOCKIO
+#define CONFIG_BCMA_BLOCKIO 1
+#endif /* CONFIG_BCMA_BLOCKIO */ 
+#ifndef CONFIG_BCMA_HOST_PCI
+#define CONFIG_BCMA_HOST_PCI 1
+#endif /* CONFIG_BCMA_HOST_PCI */ 
+#ifndef CONFIG_B43_BCMA
+#define CONFIG_B43_BCMA 1
+#endif /* CONFIG_B43_BCMA */ 
+#ifndef CONFIG_B43_BCMA_PIO
+#define CONFIG_B43_BCMA_PIO 1
+#endif /* CONFIG_B43_BCMA_PIO */ 
+#ifndef CONFIG_P54_PCI
+#define CONFIG_P54_PCI 1
+#endif /* CONFIG_P54_PCI */ 
+#ifndef CONFIG_B44
+#define CONFIG_B44 1
+#endif /* CONFIG_B44 */ 
+#ifndef CONFIG_B44_PCI
+#define CONFIG_B44_PCI 1
+#endif /* CONFIG_B44_PCI */ 
+#ifndef CONFIG_RTL8180
+#define CONFIG_RTL8180 1
+#endif /* CONFIG_RTL8180 */ 
+#ifndef CONFIG_ADM8211
+#define CONFIG_ADM8211 1
+#endif /* CONFIG_ADM8211 */ 
+#ifndef CONFIG_RT2X00_LIB_PCI
+#define CONFIG_RT2X00_LIB_PCI 1
+#endif /* CONFIG_RT2X00_LIB_PCI */ 
+#ifndef CONFIG_RT2400PCI
+#define CONFIG_RT2400PCI 1
+#endif /* CONFIG_RT2400PCI */ 
+#ifndef CONFIG_RT2500PCI
+#define CONFIG_RT2500PCI 1
+#endif /* CONFIG_RT2500PCI */ 
+#if defined(CONFIG_CRC_CCITT) || defined(CONFIG_CRC_CCITT_MODULE)
+#ifndef CONFIG_RT2800PCI
+#define CONFIG_RT2800PCI 1
+#endif /* CONFIG_RT2800PCI */ 
+#ifndef CONFIG_RT2800PCI_RT33XX
+#define CONFIG_RT2800PCI_RT33XX 1
+#endif /* CONFIG_RT2800PCI_RT33XX */ 
+#ifndef CONFIG_RT2800PCI_RT35XX
+#define CONFIG_RT2800PCI_RT35XX 1
+#endif /* CONFIG_RT2800PCI_RT35XX */ 
+#endif /* CONFIG_CRC_CCITT */
+#if defined(CONFIG_CRC_ITU_T) || defined(CONFIG_CRC_ITU_T_MODULE)
+#ifndef CONFIG_RT61PCI
+#define CONFIG_RT61PCI 1
+#endif /* CONFIG_RT61PCI */ 
+#endif /* CONFIG_CRC_ITU_T */
+#ifndef CONFIG_MWL8K
+#define CONFIG_MWL8K 1
+#endif /* CONFIG_MWL8K */ 
+#ifndef CONFIG_ATL1
+#define CONFIG_ATL1 1
+#endif /* CONFIG_ATL1 */ 
+#ifndef CONFIG_ATL2
+#define CONFIG_ATL2 1
+#endif /* CONFIG_ATL2 */ 
+#ifndef CONFIG_ATL1E
+#define CONFIG_ATL1E 1
+#endif /* CONFIG_ATL1E */ 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,27))
+#ifndef CONFIG_ATL1C
+#define CONFIG_ATL1C 1
+#endif /* CONFIG_ATL1C */ 
+#undef CONFIG_ALX
+#else /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#undef CONFIG_ATL1C
+#ifndef CONFIG_ALX
+#define CONFIG_ALX 1
+#endif /* CONFIG_ALX */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#if defined(CONFIG_WIRELESS_EXT) || defined(CONFIG_WIRELESS_EXT_MODULE)
+#ifndef CONFIG_HERMES
+#define CONFIG_HERMES 1
+#endif /* CONFIG_HERMES */ 
+#ifndef CONFIG_HERMES_CACHE_FW_ON_INIT
+#define CONFIG_HERMES_CACHE_FW_ON_INIT 1
+#endif /* CONFIG_HERMES_CACHE_FW_ON_INIT */ 
+#if defined(CONFIG_PPC_PMAC) || defined(CONFIG_PPC_PMAC_MODULE)
+#ifndef CONFIG_APPLE_AIRPORT
+#define CONFIG_APPLE_AIRPORT 1
+#endif /* CONFIG_APPLE_AIRPORT */ 
+#endif /* CONFIG_PPC_PMAC */
+#ifndef CONFIG_PLX_HERMES
+#define CONFIG_PLX_HERMES 1
+#endif /* CONFIG_PLX_HERMES */ 
+#ifndef CONFIG_TMD_HERMES
+#define CONFIG_TMD_HERMES 1
+#endif /* CONFIG_TMD_HERMES */ 
+#ifndef CONFIG_NORTEL_HERMES
+#define CONFIG_NORTEL_HERMES 1
+#endif /* CONFIG_NORTEL_HERMES */ 
+#ifndef CONFIG_PCI_HERMES
+#define CONFIG_PCI_HERMES 1
+#endif /* CONFIG_PCI_HERMES */ 
+#if defined(CONFIG_PCMCIA) || defined(CONFIG_PCMCIA_MODULE)
+#ifndef CONFIG_PCMCIA_HERMES
+#define CONFIG_PCMCIA_HERMES 1
+#endif /* CONFIG_PCMCIA_HERMES */ 
+#ifndef CONFIG_PCMCIA_SPECTRUM
+#define CONFIG_PCMCIA_SPECTRUM 1
+#endif /* CONFIG_PCMCIA_SPECTRUM */ 
+#endif /* CONFIG_PCMCIA */
+#endif /* CONFIG_WIRELESS_EXT */
+#ifndef CONFIG_RTL8192CE
+#define CONFIG_RTL8192CE 1
+#endif /* CONFIG_RTL8192CE */ 
+#ifndef CONFIG_RTL8192SE
+#define CONFIG_RTL8192SE 1
+#endif /* CONFIG_RTL8192SE */ 
+#ifndef CONFIG_RTL8192DE
+#define CONFIG_RTL8192DE 1
+#endif /* CONFIG_RTL8192DE */ 
+#ifndef CONFIG_BRCMSMAC
+#define CONFIG_BRCMSMAC 1
+#endif /* CONFIG_BRCMSMAC */ 
+#ifndef CONFIG_MWIFIEX_PCIE
+#define CONFIG_MWIFIEX_PCIE 1
+#endif /* CONFIG_MWIFIEX_PCIE */ 
+#endif /* CONFIG_PCI */
+#if defined(CONFIG_PCMCIA) || defined(CONFIG_PCMCIA_MODULE)
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,27))
+#undef CONFIG_LIBERTAS
+#undef CONFIG_LIBERTAS_CS
+#else /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#ifndef CONFIG_LIBERTAS_CS
+#define CONFIG_LIBERTAS_CS 1
+#endif /* CONFIG_LIBERTAS_CS */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#endif /* CONFIG_PCMCIA */
+#ifndef CONFIG_EEPROM_93CX6
+#define CONFIG_EEPROM_93CX6 1
+#endif /* CONFIG_EEPROM_93CX6 */ 
+#if defined(CONFIG_USB) || defined(CONFIG_USB_MODULE)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29))
+#ifndef CONFIG_COMPAT_ZD1211RW
+#define CONFIG_COMPAT_ZD1211RW 1
+#endif /* CONFIG_COMPAT_ZD1211RW */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_29 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,29))
+#undef CONFIG_USB_COMPAT_USBNET
+#undef CONFIG_USB_NET_COMPAT_RNDIS_HOST
+#undef CONFIG_USB_NET_COMPAT_RNDIS_WLAN
+#undef CONFIG_USB_NET_COMPAT_CDCETHER
+#else /* CONFIG_COMPAT_KERNEL_2_6_29 */
+#ifndef CONFIG_USB_COMPAT_USBNET
+#define CONFIG_USB_COMPAT_USBNET 1
+#endif /* CONFIG_USB_COMPAT_USBNET */ 
+#if defined(CONFIG_USB_NET_CDCETHER) || defined(CONFIG_USB_NET_CDCETHER_MODULE)
+#ifndef CONFIG_USB_NET_COMPAT_RNDIS_HOST
+#define CONFIG_USB_NET_COMPAT_RNDIS_HOST 1
+#endif /* CONFIG_USB_NET_COMPAT_RNDIS_HOST */ 
+#ifndef CONFIG_USB_NET_COMPAT_RNDIS_WLAN
+#define CONFIG_USB_NET_COMPAT_RNDIS_WLAN 1
+#endif /* CONFIG_USB_NET_COMPAT_RNDIS_WLAN */ 
+#endif /* CONFIG_USB_NET_CDCETHER */
+#if defined(CONFIG_USB_NET_CDCETHER_MODULE) || defined(CONFIG_USB_NET_CDCETHER_MODULE_MODULE)
+#ifndef CONFIG_USB_NET_COMPAT_RNDIS_HOST
+#define CONFIG_USB_NET_COMPAT_RNDIS_HOST 1
+#endif /* CONFIG_USB_NET_COMPAT_RNDIS_HOST */ 
+#ifndef CONFIG_USB_NET_COMPAT_RNDIS_WLAN
+#define CONFIG_USB_NET_COMPAT_RNDIS_WLAN 1
+#endif /* CONFIG_USB_NET_COMPAT_RNDIS_WLAN */ 
+#endif /* CONFIG_USB_NET_CDCETHER */
+#ifndef CONFIG_USB_NET_COMPAT_CDCETHER
+#define CONFIG_USB_NET_COMPAT_CDCETHER 1
+#endif /* CONFIG_USB_NET_COMPAT_CDCETHER */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_29 */
+#ifndef CONFIG_P54_USB
+#define CONFIG_P54_USB 1
+#endif /* CONFIG_P54_USB */ 
+#ifndef CONFIG_RTL8187
+#define CONFIG_RTL8187 1
+#endif /* CONFIG_RTL8187 */ 
+#if defined(CONFIG_MAC80211_LEDS) || defined(CONFIG_MAC80211_LEDS_MODULE)
+#ifndef CONFIG_RTL8187_LEDS
+#define CONFIG_RTL8187_LEDS 1
+#endif /* CONFIG_RTL8187_LEDS */ 
+#endif /* CONFIG_MAC80211_LEDS */
+#ifndef CONFIG_AT76C50X_USB
+#define CONFIG_AT76C50X_USB 1
+#endif /* CONFIG_AT76C50X_USB */ 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29))
+#ifndef CONFIG_CARL9170
+#define CONFIG_CARL9170 1
+#endif /* CONFIG_CARL9170 */ 
+#if defined(CONFIG_MAC80211_LEDS) || defined(CONFIG_MAC80211_LEDS_MODULE)
+#ifndef CONFIG_CARL9170_LEDS
+#define CONFIG_CARL9170_LEDS 1
+#endif /* CONFIG_CARL9170_LEDS */ 
+#endif /* CONFIG_MAC80211_LEDS */
+#ifndef CONFIG_CARL9170_WPC
+#define CONFIG_CARL9170_WPC 1
+#endif /* CONFIG_CARL9170_WPC */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_29 */
+#ifndef CONFIG_COMPAT_USB_URB_THREAD_FIX
+#define CONFIG_COMPAT_USB_URB_THREAD_FIX 1
+#endif /* CONFIG_COMPAT_USB_URB_THREAD_FIX */ 
+#ifndef CONFIG_ATH9K_HTC
+#define CONFIG_ATH9K_HTC 1
+#endif /* CONFIG_ATH9K_HTC */ 
+#ifndef CONFIG_RT2500USB
+#define CONFIG_RT2500USB 1
+#endif /* CONFIG_RT2500USB */ 
+#if defined(CONFIG_CRC_CCITT) || defined(CONFIG_CRC_CCITT_MODULE)
+#ifndef CONFIG_RT2800USB
+#define CONFIG_RT2800USB 1
+#endif /* CONFIG_RT2800USB */ 
+#ifndef CONFIG_RT2800USB_RT33XX
+#define CONFIG_RT2800USB_RT33XX 1
+#endif /* CONFIG_RT2800USB_RT33XX */ 
+#ifndef CONFIG_RT2800USB_RT35XX
+#define CONFIG_RT2800USB_RT35XX 1
+#endif /* CONFIG_RT2800USB_RT35XX */ 
+#ifndef CONFIG_RT2800USB_UNKNOWN
+#define CONFIG_RT2800USB_UNKNOWN 1
+#endif /* CONFIG_RT2800USB_UNKNOWN */ 
+#endif /* CONFIG_CRC_CCITT */
+#ifndef CONFIG_RT2X00_LIB_USB
+#define CONFIG_RT2X00_LIB_USB 1
+#endif /* CONFIG_RT2X00_LIB_USB */ 
+#if defined(CONFIG_CRC_ITU_T) || defined(CONFIG_CRC_ITU_T_MODULE)
+#ifndef CONFIG_RT73USB
+#define CONFIG_RT73USB 1
+#endif /* CONFIG_RT73USB */ 
+#endif /* CONFIG_CRC_ITU_T */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,27))
+#undef CONFIG_LIBERTAS_THINFIRM_USB
+#undef CONFIG_LIBERTAS_USB
+#else /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#ifndef CONFIG_LIBERTAS_THINFIRM_USB
+#define CONFIG_LIBERTAS_THINFIRM_USB 1
+#endif /* CONFIG_LIBERTAS_THINFIRM_USB */ 
+#ifndef CONFIG_LIBERTAS_USB
+#define CONFIG_LIBERTAS_USB 1
+#endif /* CONFIG_LIBERTAS_USB */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#ifndef CONFIG_ORINOCO_USB
+#define CONFIG_ORINOCO_USB 1
+#endif /* CONFIG_ORINOCO_USB */ 
+#ifndef CONFIG_BT_HCIBTUSB
+#define CONFIG_BT_HCIBTUSB 1
+#endif /* CONFIG_BT_HCIBTUSB */ 
+#ifndef CONFIG_BT_HCIBCM203X
+#define CONFIG_BT_HCIBCM203X 1
+#endif /* CONFIG_BT_HCIBCM203X */ 
+#ifndef CONFIG_BT_HCIBPA10X
+#define CONFIG_BT_HCIBPA10X 1
+#endif /* CONFIG_BT_HCIBPA10X */ 
+#ifndef CONFIG_BT_HCIBFUSB
+#define CONFIG_BT_HCIBFUSB 1
+#endif /* CONFIG_BT_HCIBFUSB */ 
+#ifndef CONFIG_BT_ATH3K
+#define CONFIG_BT_ATH3K 1
+#endif /* CONFIG_BT_ATH3K */ 
+#ifndef CONFIG_RTL8192CU
+#define CONFIG_RTL8192CU 1
+#endif /* CONFIG_RTL8192CU */ 
+#endif /* CONFIG_USB */
+#if defined(CONFIG_SPI_MASTER) || defined(CONFIG_SPI_MASTER_MODULE)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25))
+#if defined(CONFIG_CRC7) || defined(CONFIG_CRC7_MODULE)
+#ifndef CONFIG_WL1251_SPI
+#define CONFIG_WL1251_SPI 1
+#endif /* CONFIG_WL1251_SPI */ 
+#ifndef CONFIG_WL12XX_SPI
+#define CONFIG_WL12XX_SPI 1
+#endif /* CONFIG_WL12XX_SPI */ 
+#endif /* CONFIG_CRC7 */
+#ifndef CONFIG_P54_SPI
+#define CONFIG_P54_SPI 1
+#endif /* CONFIG_P54_SPI */ 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,27))
+#undef CONFIG_LIBERTAS_SPI
+#else /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#ifndef CONFIG_LIBERTAS_SPI
+#define CONFIG_LIBERTAS_SPI 1
+#endif /* CONFIG_LIBERTAS_SPI */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#endif /* CONFIG_COMPAT_KERNEL_2_6_25 */
+#endif /* CONFIG_SPI_MASTER */
+#if defined(CONFIG_MMC) || defined(CONFIG_MMC_MODULE)
+#ifndef CONFIG_SSB_SDIOHOST
+#define CONFIG_SSB_SDIOHOST 1
+#endif /* CONFIG_SSB_SDIOHOST */ 
+#ifndef CONFIG_B43_SDIO
+#define CONFIG_B43_SDIO 1
+#endif /* CONFIG_B43_SDIO */ 
+#if defined(CONFIG_CRC7) || defined(CONFIG_CRC7_MODULE)
+#if defined(CONFIG_WL12XX_PLATFORM_DATA) || defined(CONFIG_WL12XX_PLATFORM_DATA_MODULE)
+#ifndef CONFIG_COMPAT_WL1251_SDIO
+#define CONFIG_COMPAT_WL1251_SDIO 1
+#endif /* CONFIG_COMPAT_WL1251_SDIO */ 
+#endif /* CONFIG_WL12XX_PLATFORM_DATA */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,32))
+#if defined(CONFIG_WL12XX_PLATFORM_DATA) || defined(CONFIG_WL12XX_PLATFORM_DATA_MODULE)
+#ifndef CONFIG_COMPAT_WL12XX_SDIO
+#define CONFIG_COMPAT_WL12XX_SDIO 1
+#endif /* CONFIG_COMPAT_WL12XX_SDIO */ 
+#endif /* CONFIG_WL12XX_PLATFORM_DATA */
+#endif /* CONFIG_COMPAT_KERNEL_2_6_32 */
+#endif /* CONFIG_CRC7 */
+#ifndef CONFIG_MWIFIEX_SDIO
+#define CONFIG_MWIFIEX_SDIO 1
+#endif /* CONFIG_MWIFIEX_SDIO */ 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,32))
+#ifndef CONFIG_COMPAT_LIBERTAS_SDIO
+#define CONFIG_COMPAT_LIBERTAS_SDIO 1
+#endif /* CONFIG_COMPAT_LIBERTAS_SDIO */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_32 */
+#ifndef CONFIG_IWM
+#define CONFIG_IWM 1
+#endif /* CONFIG_IWM */ 
+#ifndef CONFIG_BT_HCIBTSDIO
+#define CONFIG_BT_HCIBTSDIO 1
+#endif /* CONFIG_BT_HCIBTSDIO */ 
+#ifndef CONFIG_BT_MRVL_SDIO
+#define CONFIG_BT_MRVL_SDIO 1
+#endif /* CONFIG_BT_MRVL_SDIO */ 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,27))
+#ifndef CONFIG_ATH6KL
+#define CONFIG_ATH6KL 1
+#endif /* CONFIG_ATH6KL */ 
+#ifndef CONFIG_ATH6KL_SDIO
+#define CONFIG_ATH6KL_SDIO 1
+#endif /* CONFIG_ATH6KL_SDIO */ 
+#ifndef CONFIG_ATH6KL_USB
+#define CONFIG_ATH6KL_USB 1
+#endif /* CONFIG_ATH6KL_USB */ 
+#ifndef CONFIG_ATH6KL_DEBUG
+#define CONFIG_ATH6KL_DEBUG 1
+#endif /* CONFIG_ATH6KL_DEBUG */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29))
+#ifndef CONFIG_BRCMFMAC
+#define CONFIG_BRCMFMAC 1
+#endif /* CONFIG_BRCMFMAC */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_29 */
+#endif /* CONFIG_MMC */
+#ifndef CONFIG_RTLWIFI
+#define CONFIG_RTLWIFI 1
+#endif /* CONFIG_RTLWIFI */ 
+#ifndef CONFIG_RTL8192C_COMMON
+#define CONFIG_RTL8192C_COMMON 1
+#endif /* CONFIG_RTL8192C_COMMON */ 
+#ifndef CONFIG_RT2X00
+#define CONFIG_RT2X00 1
+#endif /* CONFIG_RT2X00 */ 
+#ifndef CONFIG_RT2X00_LIB
+#define CONFIG_RT2X00_LIB 1
+#endif /* CONFIG_RT2X00_LIB */ 
+#ifndef CONFIG_RT2800_LIB
+#define CONFIG_RT2800_LIB 1
+#endif /* CONFIG_RT2800_LIB */ 
+#ifndef CONFIG_RT2X00_LIB_FIRMWARE
+#define CONFIG_RT2X00_LIB_FIRMWARE 1
+#endif /* CONFIG_RT2X00_LIB_FIRMWARE */ 
+#ifndef CONFIG_RT2X00_LIB_CRYPTO
+#define CONFIG_RT2X00_LIB_CRYPTO 1
+#endif /* CONFIG_RT2X00_LIB_CRYPTO */ 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,25))
+#undef CONFIG_RT2X00_LIB_LEDS
+#else /* CONFIG_COMPAT_KERNEL_2_6_25 */
+#if defined(CONFIG_LEDS_CLASS) || defined(CONFIG_LEDS_CLASS_MODULE)
+#ifndef CONFIG_RT2X00_LIB_LEDS
+#define CONFIG_RT2X00_LIB_LEDS 1
+#endif /* CONFIG_RT2X00_LIB_LEDS */ 
+#endif /* CONFIG_LEDS_CLASS */
+#endif /* CONFIG_COMPAT_KERNEL_2_6_25 */
+#ifndef CONFIG_P54_COMMON
+#define CONFIG_P54_COMMON 1
+#endif /* CONFIG_P54_COMMON */ 
+#if defined(CONFIG_MAC80211_LEDS) || defined(CONFIG_MAC80211_LEDS_MODULE)
+#ifndef CONFIG_P54_LEDS
+#define CONFIG_P54_LEDS 1
+#endif /* CONFIG_P54_LEDS */ 
+#endif /* CONFIG_MAC80211_LEDS */
+#ifndef CONFIG_ATH_COMMON
+#define CONFIG_ATH_COMMON 1
+#endif /* CONFIG_ATH_COMMON */ 
+#ifndef CONFIG_BRCMUTIL
+#define CONFIG_BRCMUTIL 1
+#endif /* CONFIG_BRCMUTIL */ 
+#if defined(CONFIG_CRC7) || defined(CONFIG_CRC7_MODULE)
+#ifndef CONFIG_WL1251
+#define CONFIG_WL1251 1
+#endif /* CONFIG_WL1251 */ 
+#ifndef CONFIG_WL12XX
+#define CONFIG_WL12XX 1
+#endif /* CONFIG_WL12XX */ 
+#endif /* CONFIG_CRC7 */
+#ifndef CONFIG_MWIFIEX
+#define CONFIG_MWIFIEX 1
+#endif /* CONFIG_MWIFIEX */ 
+#if !defined(CONFIG_CORDIC) && !defined(CONFIG_CORDIC_MODULE)
+#ifndef CONFIG_COMPAT_CORDIC
+#define CONFIG_COMPAT_CORDIC 1
+#endif /* CONFIG_COMPAT_CORDIC */ 
+#endif /* CONFIG_CORDIC */
+#if !defined(CONFIG_CRC8) && !defined(CONFIG_CRC8_MODULE)
+#ifndef CONFIG_COMPAT_CRC8
+#define CONFIG_COMPAT_CRC8 1
+#endif /* CONFIG_COMPAT_CRC8 */ 
+#endif /* CONFIG_CRC8 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,27))
+#undef CONFIG_LIBERTAS
+#else /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#ifndef CONFIG_LIBERTAS_THINFIRM
+#define CONFIG_LIBERTAS_THINFIRM 1
+#endif /* CONFIG_LIBERTAS_THINFIRM */ 
+#ifndef CONFIG_LIBERTAS
+#define CONFIG_LIBERTAS 1
+#endif /* CONFIG_LIBERTAS */ 
+#ifndef CONFIG_LIBERTAS_MESH
+#define CONFIG_LIBERTAS_MESH 1
+#endif /* CONFIG_LIBERTAS_MESH */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_27 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,31))
+#ifndef CONFIG_RFKILL_BACKPORT
+#define CONFIG_RFKILL_BACKPORT 1
+#endif /* CONFIG_RFKILL_BACKPORT */ 
+#if defined(CONFIG_LEDS_TRIGGERS) || defined(CONFIG_LEDS_TRIGGERS_MODULE)
+#ifndef CONFIG_RFKILL_BACKPORT_LEDS
+#define CONFIG_RFKILL_BACKPORT_LEDS 1
+#endif /* CONFIG_RFKILL_BACKPORT_LEDS */ 
+#endif /* CONFIG_LEDS_TRIGGERS */
+#ifndef CONFIG_RFKILL_BACKPORT_INPUT
+#define CONFIG_RFKILL_BACKPORT_INPUT 1
+#endif /* CONFIG_RFKILL_BACKPORT_INPUT */ 
+#endif /* CONFIG_COMPAT_KERNEL_2_6_31 */
+#ifdef CONFIG_NET_SCHED
+#ifdef CONFIG_NETDEVICES_MULTIQUEUE
+#ifndef CONFIG_MAC80211_QOS
+#define CONFIG_MAC80211_QOS 1
+#endif /* CONFIG_MAC80211_QOS */ 
+#endif
+#endif
 #endif /* COMPAT_AUTOCONF_INCLUDED */
diff --git a/wlan/atheros/compat-wireless/include/linux/export.h b/wlan/atheros/compat-wireless/include/linux/export.h
index a0ced3e..3e69147 100644
--- a/wlan/atheros/compat-wireless/include/linux/export.h
+++ b/wlan/atheros/compat-wireless/include/linux/export.h
@@ -1,10 +1,28 @@
+/*
+ * Copyright (c) 2010-2011 Atheros Communications Inc.
+ * Copyright (c) 2011 Qualcomm Atheros, Inc.
+ * Copyright (c) 2010-2012 Freescale Semiconductor, Inc.
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
 #ifndef _COMPAT_LINUX_EXPORT_H
 #define _COMPAT_LINUX_EXPORT_H 1
 
 #include <linux/version.h>
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,0,0))
-#include_next <linux/export.h>
+
 #else
 #include <linux/module.h>
 #endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,0)) */
diff --git a/wlan/atheros/compat-wireless/scripts/check_config.sh b/wlan/atheros/compat-wireless/scripts/check_config.sh
old mode 100644
new mode 100755
diff --git a/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh b/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh
old mode 100644
new mode 100755
-- 
1.8.0

